indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,265 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from typing import Optional
+
+from flask import json
+from google.auth import crypt, jwt
+from google.auth.transport.requests import AuthorizedSession
+from google.oauth2.service_account import Credentials
+from werkzeug.exceptions import BadRequest
+
+from indico.core import signals
+from indico.core.config import config
+from indico.core.logger import Logger
+
+
+class GoogleWalletManager:
+    """Google Wallet management class."""
+
+    def __init__(self, event, registration=None):
+        self.event = event
+        self.registration = registration
+        self.settings = self.get_google_wallet_settings(self.event.category)
+        self.credentials = None
+        self.http_client = None
+        self.base_url = 'https://walletobjects.googleapis.com/walletobjects/v1'
+        self.batch_url = 'https://walletobjects.googleapis.com/batch'
+        self.class_url = f'{self.base_url}/eventTicketClass'
+        self.object_url = f'{self.base_url}/eventTicketObject'
+
+        # Set up authenticated client
+        if self.settings:
+            self.settings['google_wallet_application_credentials'] = json.loads(
+                self.settings['google_wallet_application_credentials'])
+            self.auth()
+
+    @staticmethod
+    def get_google_wallet_settings(category):
+        while category:
+            if category.google_wallet_settings:
+                return category.google_wallet_settings
+            else:
+                category = category.parent
+        return None
+
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        self.credentials = Credentials.from_service_account_info(
+            self.settings['google_wallet_application_credentials'],
+            scopes=['https://www.googleapis.com/auth/wallet_object.issuer'])
+        self.http_client = AuthorizedSession(self.credentials)
+
+    def create_class_template(self, issuer_id: str, class_suffix: str) -> dict:
+        """This method will return a dict format ticket class template."""
+        from indico.modules.categories.controllers.util import make_format_event_date_func
+        logo_url = f'{config.BASE_URL}{config.IMAGES_BASE_URL}/logo_indico.png'
+        if config.LOGO_URL:
+            logo_url = config.LOGO_URL if 'http' in config.LOGO_URL else f'{config.BASE_URL}{config.LOGO_URL}'
```

Comment:
maybe you can add a new `ABSOLUTE_LOGO_URL` property to `IndicoConfig` that contains an absolute version of the configured `LOGO_URL`?

Also, `'http' in url` is not a good check. Imagine if the logo URL is `/something/images/whateverhttpwhatever.png`...